### PR TITLE
OT-navbarToTabs  | Replaced bottomNavigation with a DefaultTabController for better navigation efficiency

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I Did:**

- Implemented a new dart file called tab.dart what uses DefaultTabController to allow a user to navigate through the app's pages

- Deprecated bottomNavigation.dart, router.dart, and routing_constants.dart as these functions aren't needed anymore

- Changed main.dart to have a home page of the login page

- Changed login.dart to redirect to the DefaultTabController

- Commented out any calls to the previous bottomNavigation.dart, namely in recordingSession.dart, and settings_page.dart

**How To Test:**

- You will need zookeeper, kafka, and spring boot running

- login into the application using any email already in your MySQL database

- You should be able to see the new DefaultTabController footer where the bottomNavigation footer used to be. It should look identical and functions should not have changed

- Test each tab: switch to Reports, then to Settings, then back to Home

### CODE REVIEW

File: lib/components/footer/bottomNavigation.dart [hunk 1]
```diff
@@ -1,35 +1,32 @@
 import 'package:flutter/material.dart';
```
Comment (frlzjosh) [inline lib/components/footer/bottomNavigation.dart]: remove this file

File: lib/settings_page.dart [hunk 2]
```diff
@@ -111,7 +111,7 @@ class _MySettingsPageState extends State<MySettingsPage> {
               visible: _showSettings,
             )
           ]))),
-      bottomNavigationBar: AppBarBuilder(),
+      //bottomNavigationBar: AppBarBuilder(),
```
Comment (frlzjosh) [inline lib/settings_page.dart]: delete this line of code

File: lib/tabs.dart [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:off_top_mobile/login.dart';
+import 'package:off_top_mobile/recordingSession.dart';
+import 'package:off_top_mobile/settings_page.dart';
+
+class Tabs extends StatelessWidget {
```
Comment (frlzjosh) [inline lib/tabs.dart]: move this to components/footer/ and consider renaming it to the something more readible like footerTabs or bottomNavigationTabs

**Review by frlzjosh — CHANGES_REQUESTED** [review]
Small things like renaming certain things and deleting some lines of code

**Comment by tonynguyen98** [comment]
Currently working on these changes, should I also remove router.dart and routing_constants.dart as they aren't used anymore?

**Comment by tonynguyen98** [comment]
>  Small things like renaming certain things and deleting some lines of code

Currently working on these changes, <del> should I also remove router.dart and routing_constants.dart as they aren't used anymore?</del>

I think we should keep router.dart and routing_constants.dart in the event where we need it for the login page implementation.

File: lib/components/footer/bottomNavigationTabs.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:off_top_mobile/login.dart';
+import 'package:off_top_mobile/recordingSession.dart';
+import 'package:off_top_mobile/settings_page.dart';
+
+class Tabs extends StatelessWidget {
```
Comment (frlzjosh) [inline lib/components/footer/bottomNavigationTabs.dart]: Have the class name match the name of the file

**Review by frlzjosh — CHANGES_REQUESTED** [review]
one small request!

File: lib/components/footer/bottomNavigationTabs.dart (hunk 4 above)
Comment (tonynguyen98) [reply lib/components/footer/bottomNavigationTabs.dart]: Done! Completely missed that, my bad.

**Review by frlzjosh — APPROVED** [review]
nice!

**Pull request merged**